cynet/client.go

```diff
@@ -0,0 +1,582 @@
+				backoff := time.Duration(retryCount) * time.Second
+				select {
+				case <-a.ctx.Done():
+					return nil, a.ctx.Err()
+				case <-time.After(backoff):
+					continue
+				}
+			}
+			return nil, fmt.Errorf("request failed: %w", err)
+		}
+
+		defer resp.Body.Close()
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read response: %w", err)
+		}
+
+		// Handle different status codes
+		switch resp.StatusCode {
+		case http.StatusOK:
+			var alertsResp AlertsResponse
+			if err := json.Unmarshal(body, &alertsResp); err != nil {
+				return nil, fmt.Errorf("failed to unmarshal alerts response: %w", err)
+			}
+			return &alertsResp, nil
+
+		case http.StatusUnauthorized:
+			// Try to refresh token once
+			a.conf.ClientOptions.OnWarning("received 401, attempting to refresh token")
+			if err := a.authenticate(); err != nil {
+				return nil, fmt.Errorf("re-authentication failed: %w", err)
+			}
+			// Retry the request with new token
+			retryCount++
+			continue
+
+		case http.StatusForbidden:
+			// Access forbidden - shut down adapter
+			a.conf.ClientOptions.OnError(fmt.Errorf("received 403 Forbidden - access denied to alerts API"))
+			go a.Close()
+			return nil, fmt.Errorf("access forbidden (403) - shutting down adapter")
+
+		case http.StatusTooManyRequests:
+			// Handle rate limiting
+			retryAfter := resp.Header.Get("Retry-After")
+			waitDuration := 60 * time.Second // Default wait
+
+			if retryAfter != "" {
+				if seconds, err := strconv.Atoi(retryAfter); err == nil {
+					waitDuration = time.Duration(seconds) * time.Second
+				}
+			}
+
+			a.conf.ClientOptions.OnWarning(fmt.Sprintf("rate limited (429), waiting %v before retry", waitDuration))
+
+			select {
+			case <-a.ctx.Done():
+				return nil, a.ctx.Err()
+			case <-time.After(waitDuration):
+				retryCount++
+				continue
+			}
+
+		case http.StatusInternalServerError, http.StatusBadGateway, http.StatusServiceUnavailable:
+			// Server errors - retry with exponential backoff
+			lastErr = fmt.Errorf("server error %d: %s", resp.StatusCode, string(body))
+			a.conf.ClientOptions.DebugLog(lastErr.Error())
+			retryCount++
+
+			if retryCount < maxRetries {
+				backoff := time.Duration(retryCount*retryCount) * time.Second
+				a.conf.ClientOptions.OnWarning(fmt.Sprintf("server error %d, retrying in %v (attempt %d/%d)",
+					resp.StatusCode, backoff, retryCount, maxRetries))
+
+				select {
+				case <-a.ctx.Done():
+					return nil, a.ctx.Err()
+				case <-time.After(backoff):
+					continue
+				}
+			}
+
+		default:
+			return nil, fmt.Errorf("unexpected status code %d: %s", resp.StatusCode, string(body))
+		}
+	}
+
+	if lastErr != nil {
+		return nil, fmt.Errorf("request failed after %d retries: %w", maxRetries, lastErr)
+	}
+
+	return nil, fmt.Errorf("request failed after %d retries", maxRetries)
+}
+
+func (a *CynetAdapter) processAlerts(resp *AlertsResponse) ([]utils.Dict, time.Time) {
+	var newAlerts []utils.Dict
```

Comment:
```suggestion
	cutoff := time.Now().Add(-2 * time.Duration(queryInterval) * time.Second).Unix()
```
`queryInterval` is an untyped int, so `Add(-1 * queryInterval)` adds nanoseconds, not seconds. Also changed to use `time.Now()` since dedupe entries store wall-clock time, not event time.